### ISSUE TITLE
Getting error 400 with sample from README

### ISSUE THREAD

**Issue by matejzero** [broadinstitute/python-cert_manager#28]
Hey,

I'm trying to create a cert from the sample code in README, but I'm getting a 400 error back.

Code:
```python
  result = ssl.enroll(
      cert_type_name="our_cert_type", csr=csr, term=365, org_id=cert_org[0]["id"]
  )
```

Error:
```
Traceback (most recent call last):
  File "/Users/matejz/certificates/generate_cert_from_csr.py", line 37, in <module>
    result = ssl.enroll(
  File "/usr/local/lib/python3.9/site-packages/cert_manager/_certificates.py", line 142, in enroll
    result = self._client.post(url, data=data)
  File "/usr/local/lib/python3.9/site-packages/cert_manager/_helpers.py", line 75, in log_traffic
    raise herr
  File "/usr/local/lib/python3.9/site-packages/cert_manager/_helpers.py", line 67, in log_traffic
    result = func(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/cert_manager/client.py", line 157, in post
    result.raise_for_status()
  File "/usr/local/lib/python3.9/site-packages/requests/models.py", line 943, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: 400 for url: https://cert-manager.com/api/ssl/v1/enroll
```

According to Sectigo API documentation, response body should include an error providing further information. Should that be included in the exception? Do you return response content as well?

**Comment by matejzero** [comment]
After some testing, it seems like you don't.

In case of errors, the API returns the code and reason for error (https://sectigo.com/uploads/audio/Certificate-Manager-20.1-Rest-API.html#overview-errors) which is really useful for debugging, but `raise_for_status()` hides it.

**Comment by evilensky** [comment]
>raise_for_status() hides it.

From the [requests docs](https://2.python-requests.org/en/master/user/quickstart/)

>It should be noted that the success of the call to r.json() does not indicate the success of the response. Some servers may return a JSON object in a failed response (e.g. error details with HTTP 500). Such JSON will be decoded and returned. To check that a request is successful, use r.raise_for_status() or check r.status_code is what you expect.

Sounds like it should be possible to add the contents of Sectigo's error to the Exception message?

The `raise` is sprinkled throughout each method itself; maybe it's prudent to extract that into its own method which can collect the sectigo message and then `raise` ?

https://github.com/broadinstitute/python-cert_manager/blob/main/cert_manager/client.py#L157-L160 

Other clients I've worked with have featured a `validate` or maybe `validate_json` method that checks outputs from `result` in just such a way, rather than returning `result` directly in each method.

**Comment by matejzero** [comment]
Yes, I think that would be a good solution. 

I sometimes return dict with `ret (bool)` and `msg (string)` fields and it works OK in my cases.

**Comment by coreone** [comment]
Sorry it took a while to get back to this.  The `response.raise_for_status()` method will raise a `requests.exceptions.HTTPError` exception, which also stores the response itself inside the exception.  Therefore, if you wrap calls to the function in a `try` block, you should still be able to access the JSON returned:

```python
from requests.exceptions import HTTPError

try:
    result = ssl.enroll(
      cert_type_name="our_cert_type", csr=csr, term=365, org_id=cert_org[0]["id"]
  )
except HTTPError as herr:
    print(herr.response.json())
```

**Comment by titansmc** [comment]
I am receiving the same error with the sample, I don't get what is wrong here, could someone elaborate a bit more, please?

**Comment by coreone** [comment]
Sorry about the confusion on this. The way you pass the CSR in the `enroll` function was incorrect in the README. It has been corrected in #94
